### PR TITLE
multi cluster: Add replicas, nodeSelector, tolerations, affinity & resources to helm chart

### PR DESCRIPTION
**Pull request by Monkman08**
Support replicas, nodeSelector, affinity, resources and tolerations for linkerd-multicluster Chart.

There is support to choose nodes for linkerd and linkerd-viz charts but no support for linkerd-multicluster chart.

Properties replicas, nodeSelector, affinity, resources and tolerations are added to values.yml and processed in templates in the chart.

### CODE REVIEW

File: multicluster/charts/linkerd-multicluster/templates/gateway.yaml [hunk 1]
```diff
@@ -32,6 +32,20 @@ spec:
       containers:
         - name: pause
           image: gcr.io/google_containers/pause
+          resources:
+            {{- toYaml .Values.gateway.resources | nindent 12 }}
```
Comment (alpeb) [inline multicluster/charts/linkerd-multicluster/templates/gateway.yaml:36]: Can you wrap with `with` like you did below, so we don't end up with an empty section here with `.Values.gateway.resources` is not declared?

**Review by alpeb — COMMENTED** [review]
Thanks @Monkman08 :+1: 
Do you mind adding some small comment descriptions to the new `values.yaml` entries? Then, you'd have to run `bin/helm-docs` to re-generate the chart `README.md` with those descriptions.

**Comment by alpeb** [comment]
Thanks for the latest updates @Monkman08 

I had forgotten changes are also required in `multicluster/values/values.go` adding entries for the new values you created. This is needed so that the CLI interface for installing multicluster is also aware of those new values and so the templates it generates don't end with those values empty.

**Review by alpeb — APPROVED** [review]
Thanks for the prompt turnaround @Monkman08, this looks good to me! :medal_military:

**Review by kleimkuhler — APPROVED** [review]
Thanks @Monkman08!

**Review by kleimkuhler — COMMENTED** [review]
In order to pass the markdown lints, you'll need to add contexts to the markdown links in the `values.yaml`.

For example:

```
  # -- Node Selector settings for the gateway pods. (https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector)
```

Should be:

```
  # -- [Node Selector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) settings for the gateway pods.
```

**Comment by Pothulapati** [comment]
Hey @Monkman08, Looks like there is one small change to make the CI happy i.e not using bare URL's in markdown i.e https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md . Can you update the links added to the `values.yaml` to also have [] with the description? Can you also re-generate the readme once that is done?

Looking forward to merge this PR! :)

**Comment by alpeb** [comment]
The integration test is failing because the `Resources` field is being rendered using the `Resources` struct:
```go
type Resources struct {
	CPU    Constraints `json:"cpu"`
	Memory Constraints `json:"memory"`
}
```
But kubernetes expects something like
```yaml
 limits:
   cpu: 100m
   memory: 128Mi
 requests:
   cpu: 100m
   memory: 128Mi
```
I think the best approach should be to follow what we did with the main chart and make use of `partials.resources`:
https://github.com/linkerd/linkerd2/blob/188352a2054589e79f7c0f0fe1fdbc2cb94f3fcf/charts/linkerd2/templates/destination.yaml#L167-L169

**Comment by kleimkuhler** [comment]
Hey @Monkman08 just checking in to see if you need anything to get this wrapped up? Happy to help however we can!

**Comment by alpeb** [comment]
Closing this one for lack of activity. @Monkman08 feel free to re-open if you're still interested :slightly_smiling_face:

**Pull request closed without merging**